src/monodroid/jni/android-system.cc

```diff
@@ -226,6 +226,41 @@ AndroidSystem::monodroid_get_system_property (const char *name, char **value)
 	return len;
 }
 
+int
+AndroidSystem::monodroid_read_file_into_memory (const char *path, char **value)
+{
+	int i;
+
+	if (value)
+		*value = NULL;
+
+	FILE* fp = utils.monodroid_fopen (path, "r");
+	if (fp == NULL)
+		return 0;
+
+	struct stat fileStat;
+	if (fstat (fileno (fp), &fileStat) < 0) {
+		fclose (fp);
+		return 0;
+	}
+
+	if (!value) {
+		fclose (fp);
```

Comment:
Instead of repeated cleanup logic, as we're not dealing with too many independent variables, indentation might result in something saner:

```cc
int r = 0;
FILE *fp = utils.monodroid_fopen (path, "r");
if (fp != NULL) {
    stat fileStat;
    if (fstat (fileno (fp), &fileStat) == 0) {
        r = fileStat.st_size+1;
        if (value && (*value = malloc (r))) {
            fread (*value, 1, fileStat.st_size, fp);
        }
    }
    close (fp);
}
return r;
```

I find this mentally easier to reason about, as I don't need to track that every codepath actually closes `fp`.  (*Or*, this is a reason to use `std::stream`, and let destructors sort it out. ;-)